### PR TITLE
[Messenger] Fix PHPDoc template for `TransportFactoryInterface`

### PR DESCRIPTION
**Pull request by Ferror**
Improved a little bit phpdocs when dealing with Transport & Transport Factory Interface.

### CODE REVIEW

File: src/Symfony/Component/Messenger/Transport/TransportFactoryInterface.php [hunk 1]
```diff
@@ -17,9 +17,14 @@
  * Creates a Messenger transport.
  *
  * @author Samuel Roze <samuel.roze@gmail.com>
+ *
+ * @template T
```
Comment (derrabus) [inline src/Symfony/Component/Messenger/Transport/TransportFactoryInterface.php]: Right now, `T` would be of type mixed, which is _a little_ too wide. Moreover, we should use a more expressive template variable name and allow covariance.

```suggestion
 * @template-covariant TTransport of TransportInterface
```

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/AmazonSqsTransport.php [hunk 2]
```diff
@@ -78,10 +81,7 @@ public function setup(): void
         }
     }
 
-    /**
-     * @return void
-     */
-    public function reset()
+    public function reset(): void
```
Comment (derrabus) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/AmazonSqsTransport.php]: That's a BC break. We can't do that in a minor release.

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/AmazonSqsTransport.php [hunk 3]
```diff
@@ -44,6 +44,9 @@ public function __construct(Connection $connection, SerializerInterface $seriali
         $this->sender = $sender;
     }
 
+    /**
+     * @inheritDoc
+     */
```
Comment (derrabus) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/AmazonSqsTransport.php]: This block serves no purpose. Let's remove it?

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/AmazonSqsTransport.php [hunk 4]
```diff
@@ -90,10 +90,7 @@ public function reset()
         }
     }
 
-    /**
-     * @return MessageCountAwareInterface&ReceiverInterface
-     */
-    private function getReceiver(): ReceiverInterface
+    private function getReceiver(): MessageCountAwareInterface&ReceiverInterface
```
Comment (derrabus) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/AmazonSqsTransport.php]: Let's postpone this change to 7.0 where we can use this type for the constructor parameter and the property as well?

**Review by derrabus — CHANGES_REQUESTED** [review]
If you make the `TransportFactoryInterface` generic, you need to update the PHPDoc on every implementation accordingly.

**Comment by Ferror** [comment]
> If you make the `TransportFactoryInterface` generic, you need to update the PHPDoc on every implementation accordingly.

Did as requested :)

**Comment by Ferror** [comment]
@derrabus WDYT?

**Review by derrabus — APPROVED** [review]

**Review by kbond — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Ferror.

**Comment by ro0NL** [comment]
wasnt updating the returntypes on `createTransport` sufficient? eg. `SchedulerTransportFactory`

**Comment by Ferror** [comment]
> wasnt updating the returntypes on `createTransport` sufficient? eg. `SchedulerTransportFactory`

Unfortunately not. SchedulerTransportFactory shares Transport Interface therefore by pipeline config it was also required to change to add that.

**Comment by ro0NL** [comment]
i mean why prefer templating a factory when the return type is already sufficient?

https://github.com/symfony/symfony/blob/2bac801f8a4cba02e8ecc840fa7bf512034f860a/src/Symfony/Component/Scheduler/Messenger/SchedulerTransportFactory.php#L34

or what does this PR solve? :thinking:

**Comment by stof** [comment]
If we never benefit from the generic types in places in which we inject a TransportFactory but only in places using a concrete factory explicitly, we could indeed avoid making it generic but using a refined return type...

**Comment by ro0NL** [comment]
i guess we do :rocket:

**Pull request merged**